document/controller.py

```diff
@@ -0,0 +1,121 @@
+from abc import ABC, abstractmethod
+from typing import Annotated
+
+from fastapi import Body, File, Form, HTTPException, UploadFile
+from loguru import logger
+from document import DocumentFilter, DocumentCreate, DocumentUpdate
+from document.document import Document, DocumentTitleError, DocumentTypeError, ObjectNameError
+from document.service import DocumentService
+
+
+class DocumentController(ABC):
+    @abstractmethod
+    def fetch_documents(self, filter: DocumentFilter):
+        pass
+      
+    @abstractmethod
+    def fetch_document_by_name(self, object_name: str):
+        pass
+      
+    @abstractmethod
+    def fetch_document_by_id(self, doc_id: str):
+        pass
+
+    @abstractmethod
+    def upload_document(self, request: DocumentCreate):
+        pass
+
+    @abstractmethod
+    def update_document(self, doc_id: str, request: DocumentUpdate):
+        pass
+    
+    @abstractmethod
+    def delete_document(self, doc_id: str):
+        pass
+      
+class DocumentControllerV1(ABC):
+    def __init__(self, service: DocumentService):
+        self.service = service
+        self.logger = logger.bind(service="DocumentController")
+        super().__init__()
+        
+    def fetch_documents(
+      self, 
+      id: str | None = None,
+      object_name: str | None = None,
+      created_after: str | None = None,
+      created_before: str | None = None,
+      updated_after: str | None = None,
+      updated_before: str | None = None,
+    ):
+        doc_filter = DocumentFilter(
+          id = id,
+          object_name = object_name,
+          created_after = created_after,
+          created_before = created_before,
+          updated_after = updated_after,
+          updated_before = updated_before,
+        )
+        try:
+            docs = self.service.get_documents(filter=doc_filter)
+            if docs is None:
+                return HTTPException(status_code=404)
+            return docs
+        except Exception as e:
+            self.logger.error(e.args)
+            return HTTPException(status_code=500)
```

Comment:
It’s better to use raise to maintain consistency with how it's used in other modules.

Reply:
thanks for the detailed PR review jason, will fix this soon!